packages/esm-patient-appointments-app/src/appointments/appointments-form.component.tsx

```diff
@@ -1,279 +1,230 @@
   const { t } = useTranslation();
-  const history = useHistory();
-
-  useEffect(() => {
-    // toggle based on required fields
-    if (appointmentDate && appointmentStartTime && appointmentEndTime) {
-      setEnableCreateButtons(true);
-    }
-  }, [appointmentDate, appointmentStartTime, appointmentEndTime]);
-
-  useEffect(() => {
-    if (session && !currentSession && !location) {
-      setCurrentSession(session);
-      setLocation(session?.sessionLocation?.uuid);
-    }
-  }, [session, currentSession, location]);
-
-  useEffect(() => {
-    const abortController = new AbortController();
-
-    if (patientUuid) {
-      getAppointmentServiceAll(abortController).then(({ data }) => {
-        setAppointmentService(data);
-      }, createErrorHandler());
-    }
-
-    return () => abortController.abort();
-  }, [patientUuid]);
-
-  useEffect(() => {
-    const abortController = new AbortController();
-    if (serviceUuid && serviceUuid !== 'default') {
-      getAppointmentService(abortController, serviceUuid).then(({ data }) => {
-        setAppointmentServiceType(data.serviceTypes);
-      });
-    }
-  }, [serviceUuid]);
-
-  const navigate = () => {
-    history.push(`/patient/${patientUuid}/chart/appointments`);
-  };
-
-  const closeForm = (event: SyntheticEvent<HTMLButtonElement>) => {
-    formRef.current.reset();
-    let userConfirmed: boolean = false;
-    if (formChanged) {
-      userConfirmed = confirm('There is ongoing work, are you sure you want to close this tab?');
-    }
+  const locations = useLocations();
+  const session = useSessionUser();
+  const { status, services, error } = useAppointmentService();
 
-    if (userConfirmed && formChanged) {
-      entryCancelled();
-      closeComponent();
-    } else if (!formChanged) {
-      entryCancelled();
-      closeComponent();
-    }
-  };
+  const [formData, setFormData] = useState<AppointmentFormData>({
+    timeFormat: new Date().getHours() >= 12 ? 'PM' : 'AM',
+    startDate: new Date().toISOString(),
+    startTime: dayjs(new Date()).format('hh:mm'),
+  });
 
-  const handleCreateFormSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    let startDateTime = new Date(appointmentDate + ' ' + appointmentStartTime);
-    let endDateTime = new Date(appointmentDate + ' ' + appointmentEndTime);
+  const serviceTypes = useMemo(
+    () => services.find((service) => service.uuid === formData?.serviceUuid)?.serviceTypes ?? [],
+    [services, formData],
+  );
 
-    let appointment: Appointment = {
-      serviceTypeUuid: serviceTypeUuid,
+  const handleSubmit = () => {
+    const { startDate, timeFormat, startTime, serviceType, serviceUuid, comments, locationUuid } = formData;
+    const [hours, minutes] = convertTime12to24(startTime, timeFormat);
+    const startDatetime = new Date(
+      dayjs(startDate).year(),
+      dayjs(startDate).month(),
+      dayjs(startDate).date(),
+      hours,
+      minutes,
+    );
+    dayjs();
```

Comment:
Not a dayjs expert: Is this required or left in by accident?

Reply:
This was left by accident